pandas/_libs/tslibs/period.pyx

```diff
@@ -1191,12 +1191,15 @@ cdef int64_t period_ordinal_to_dt64(int64_t ordinal, int freq) except? -1:
     return dtstruct_to_dt64(&dts)
 
 
-def period_format(int64_t value, int freq, object fmt=None):
+cdef str period_format(int64_t value, int freq, object fmt=None):
```

Comment:
Not sure the extent of Cython annotation support but `Optional[Union[str, bytes]]` would be nice for `fmt` here if possible

Reply:
> Cython annotation support but Optional[Union[str, bytes]]

I think that can be done, but it wouldn't resemble the pure-py annotation at all, so would be more confusing than helpful.  In the next pass I'll see if I can get rid of the str cases, at which point it will be easier